shared_libraries/orca_shared/recovery/shared_recovery.py

```diff
@@ -191,18 +191,30 @@ def post_entry_to_queue(
     md5_body = hashlib.md5(body.encode("utf8")).hexdigest()  # nosec
 
     LOGGER.debug("Sending message to the QUEUE")
-    response = mysqs.send_message(
-        QueueUrl=db_queue_url,
-        MessageDeduplicationId=deduplication_id,
-        MessageGroupId="request_files",
-        MessageAttributes={
-            "RequestMethod": {
-                "DataType": "String",
-                "StringValue": request_method.value,
-            }
-        },
-        MessageBody=body,
-    )
+    if ".fifo" in db_queue_url:
```

Comment:
Really not a fan of code in the shared_recovery being used for queues beyond the recovery queue. This type of branching should not be necessary. If we really want posting to the status queue and posting to the copy queue in the same lib, then we can separate this into two functions and use the fifo-only function without a RequestMethod for post_copy_request_to_queue.

Reply:
yes we can create a new function for fifo queue but that means adding duplicate code in the shared lib. that's why decided to use `if else `statement

Reply:
As-is deduplication_id, and request_method both live in limbo, required without actually being required for all cases. I think the fact that request_method is an unused parameter is enough to split this into two functions. You can pull 178-183 and 218-231 into shared functions, then do

```
mysqs = setup_sqs_connection(db_queue_url)

md5_body = hashlib.md5(body.encode("utf8")).hexdigest()  # nosec
LOGGER.debug("Sending message to the QUEUE")
response = mysqs.send_message(unique code isolated)

check_sqs_response_code(response_
```

Reply:
We will want eventually to completely separate non-shared code for posting to the copy queue, as it is not fully related to the status queue.

Reply:
ok makes sense. I can separate the functions. anything you want to add @bhazuka ?

Reply:
Looking over the code, this looks to be a base function. I agree it will be easier to maintain to break into two separate base functions, one for FIF queues and one for standard. This would also allow you to remove other items not needed like `MessageAttributes` I could also see where this could be split into three functions if you really wanted.
- A base SQS function that is generic that creates the client, gets the md5hash of the body and sends the message and validates using **kwargs so that it could be used generically anywhere. This is a function that could be used in a more generic utility library later.
- A function for Fifo that creates the dedupe info and adds Message Attributes, Group ID, etc then calls the base function 
- A Function for standard that just sends the QueueUrl and body

Reply:
added a new function for the standard queue. used the existing function for the fifo queue